tornjak-frontend/src/components/dashboard/dashboard-details.js

```diff
@@ -0,0 +1,223 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import { withStyles } from '@material-ui/core/styles';
+// Components
+import {
+    Container,
+    Grid,
+    Paper,
+} from '@material-ui/core';
+// Tables
+import ClustersTable from './clusters-dashboard-table';
+import AgentsTable from './agents-dashboard-table';
+import EntriesTable from './entries-dashboard-table';
+import DashboardDrawer from './dashboard-drawer';
+
+const styles = theme => ({
+    root: {
+        marginTop: -25,
+        marginLeft: -20,
+        display: 'flex',
+    },
+    appBarSpacer: theme.mixins.toolbar,
+    content: {
+        flexGrow: 1,
+    },
+    container: { //container for root
+        paddingTop: theme.spacing(4),
+        paddingBottom: theme.spacing(4),
+        marginLeft: 0
+    },
+    paper: { //container for all grids
+        padding: theme.spacing(2),
+        display: 'flex',
+        overflow: 'auto',
+        flexDirection: 'column',
+        marginBottom: 20
+    },
+    detailsTitle: {
+        fontSize: 20,
+        color: 'rgb(89, 103, 185)'
+    },
+    metadataTag: {
+        fontWeight: 'bold',
+        fontSize: 17,
+        marginTop: 10,
+    },
+
+    dashboardDetalsLine: {
+        marginLeft: 0,
+        width: 1200,
+    },
+    metadataDetails: {
+        fontSize: 17,
+    }
+});
+
+class DashboardDetails extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+        };
+    }
+
+    selectedDataKey() {
```

Comment:
Hmm, let's move this into the constructor instead and so it only needs to run once, and it can just use the values in state. Since it's just using props as the input, which it should have at construction time.